### PR TITLE
LoginisON

### PR DESCRIPTION
**Pull request by YarinMeron**
Adding the right models and methods

### CODE REVIEW

File: ShenkinStore/Models/LoginModel.cs [hunk 1]
```diff
@@ -1,11 +1,20 @@
...
 using System.Linq;
 using System.Threading.Tasks;
 
 namespace ShenkinStore.Models
 {
     public class LoginModel
     {
+        [Required(AllowEmptyStrings = false, ErrorMessage = "UserName is Requierd.")]
+        [Display(Name = "User Name")]
+
```
Comment (zeevig) [inline ShenkinStore/Models/LoginModel.cs]: Remove unnecessary break lines and between params for example:
```
        [Required(AllowEmptyStrings = false, ErrorMessage = "UserName is Requierd.")]
        [Display(Name = "User Name")]
        public string UserName { get; set; }
        
        [Required(AllowEmptyStrings = false, ErrorMessage = "Password is Requierd.")]
        [DataType(DataType.Password)]
        public string Password { get; set; }
```

File: ShenkinStore/Models/User.cs [hunk 2]
```diff
@@ -12,22 +13,32 @@ public class User
         public int UserId { get; set; }
         [Display(Name = "User Name")]
         [DataType(DataType.Text)]
-        [Required(ErrorMessage = "Please enter name"), MaxLength(20)]
+        //[Required(ErrorMessage = "Please enter name"), MaxLength(20)]
```
Comment (zeevig) [inline ShenkinStore/Models/User.cs]: Remove unnecessary comments in this file and separate the parameters as stated in LoginModel.cs comment

File: ShenkinStore/Models/User.cs [hunk 3]
```diff
@@ -12,22 +13,32 @@ public class User
...
+        [NotMapped]
         public string ConfirmedPassword { get; set; }
         [DataType(DataType.EmailAddress)]
-        [Required(ErrorMessage = "Please enter Email ID")]
+        //[Required(ErrorMessage = "Please enter Email Adress")]
+        [Required(AllowEmptyStrings = false, ErrorMessage = "Email is Requierd.")]
+
         [RegularExpression(@"^\w+([-+.']\w+)*@\w+([-.]\w+)*\.\w+([-.]\w+)*$", ErrorMessage = "Email is not valid.")]
         public string Email { get; set; }
+        [Display(Name = "Phone - (Optional)")]
```
Comment (zeevig) [inline ShenkinStore/Models/User.cs:30]: What about the phone regex?

**Review by zeevig — COMMENTED** [review]
Good job, small comments.
Fix them and let's merge it

File: ShenkinStore/Controllers/UsersController.cs [hunk 4]
```diff
@@ -167,16 +167,16 @@ public ActionResult Register(User aUser)
...
                     objuser.CreatedOn = DateTime.Now;
                     objuser.Email = aUser.Email;
                     objuser.UserName = aUser.UserName;
                     objuser.Password = aUser.Password;
                     objuser.Phone = aUser.Phone;
-                    objuser.SuccessMessege = "User is successfully add.";
                     _context.User.Add(objuser);
                     _context.SaveChanges();
-                    return View("Login");
+                    objuser = new User();
```
Comment (zeevig) [inline ShenkinStore/Controllers/UsersController.cs]: Why create a new object? this will override the first assignment.

Comment (YarinMeron) [reply ShenkinStore/Controllers/UsersController.cs]: It meant to return an empty form with the success-message to the user. i will remove it

File: ShenkinStore/Controllers/UsersController.cs [hunk 5]
```diff
@@ -217,7 +221,7 @@ public ActionResult Login(LoginModel objloginModel)
         }
         public ActionResult Logout()
         {
-            TempData.Clear();
+            HttpContext.Session.Remove("UserName");
```
Comment (zeevig) [inline ShenkinStore/Controllers/UsersController.cs]: Need to remove the UserType as well

File: ShenkinStore/Views/Shared/AccsView.cshtml [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+                    </button>
+                    <div class="navbar-collapse collapse d-sm-inline-flex flex-sm-row-reverse">
+                        <partial name="_LoginPartial" />
+                        <ul class="navbar-nav flex-grow-1">
+                            <li class="nav-item">
+                                <a class="nav-link text-dark" asp-area="" asp-controller="Home" asp-action="Index">Home</a>
+                            </li>
+                            @*
+                                <li class="nav-item">
+                                    <a class="nav-link text-dark" asp-area="" asp-controller="Users">Users</a>
```
Comment (zeevig) [inline ShenkinStore/Views/Shared/AccsView.cshtml]: Users should not appear for User role

**Review by zeevig — COMMENTED** [review]
Fix the small issues and let's merge it

**Pull request merged**